source/documentation/deploying_services/sqs.md

```diff
@@ -0,0 +1,249 @@
+cf service SERVICE_NAME
+```
+
+where `SERVICE_NAME` is a unique descriptive name for this SQS queue.
+
+### Bind an AWS SQS queue to your app
+
+You must bind the SQS queue to your app so you can get credentials to use the
+[AWS SQS API](https://docs.aws.amazon.com/cli/latest/reference/sqs/) to read
+from and write to the queue.
+
+You can bind an SQS queue using the command line or the manifest file.
+
+Run the following to bind an SQS queue to your app:
+
+```
+cf bind-service APP_NAME SERVICE_NAME
+```
+
+It will take between 1 and 5 minuites to setting the binding. To check its progress, run:
+
+```
+cf service SERVICE_NAME
+```
+
+where:
+
+- `APP_NAME` is the name of a deployed instance of your app, as specified in your app's manifest or push command
+- `SERVICE_NAME` is a unique descriptive name for this SQS queue
+
+You can use the [app's manifest](https://docs.cloud.service.gov.uk/deploying_apps.html#deploying-public-apps)
+to bind an SQS queue to the app with default read and write permissions only.
+It will bind automatically when you next deploy your app. An example manifest:
+
+```
+--
+applications:
+- name: APP_NAME
+  services:
+  - SERVICE_NAME
+```
+
+By default, you can only [connect to an SQS queue from your GOV.UK PaaS-hosted app](#connect-to-an-sqs-queue-from-your-app).
+
+To [connect to an SQS queue from outside the GOV.UK PaaS](#connect-to-an-sqs-queue-from-outside-of-the-gov-uk-paas), you must create a service key.
+
+### Connect to an SQS queue from your app
+
+Restart your app to populate your `VCAP_SERVICES` [environment variable](/deploying_apps.html#system-provided-environment-variables)
+with the credentials needed to connect to the SQS queue.
+
+An example `VCAP_SERVICES` environment variable:
+
+```
+{
+ "VCAP_SERVICES": {
+  "aws-sqs-queue": [
+   {
+    "binding_name": null,
+    "credentials": {
+     "aws_access_key_id": "XXXXXXXXXXXXXXXXXXXX",
+     "aws_region": "eu-west-2",
+     "aws_secret_access_key": "XXXX/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+     "primary_queue_url": "https://XXXXXXXXXXXXX",
+     "secondary_queue_url": "https://XXXXXXXXXXXXX"
+    },
+    "instance_name": "my-queue",
+    "label": "aws-sqs-queue",
+    "name": "my-queue",
+    "plan": "standard",
+    "provider": null,
+    "syslog_drain_url": null,
+    "tags": [
+     "sqs"
+    ],
+    "volume_mounts": []
+   }
+  ]
+ }
+}
+```
+
+You must pass credentials from your `VCAP_SERVICES` environment variable to
+your AWS software development kit (SDK). Use the appropriate [Amazon SDK
+documentation](https://aws.amazon.com/developer/tools/#sdks) to do this.
+
+You should use the provided `primary_queue_url` in AWS API calls to make use of
+the queue.
+
+Your app may also connect to a second isolated queue using the `secondary_queue_url`
+value, however we recommend ignoring this value unless you are configuring a
+dead-letter queue. See [configuring a dead letter queue][].
+
+### Connect to an SQS queue from outside of the GOV.UK PaaS
```

Comment:
Should we remove this "connect from outside" section for SQS for now? 

... I'm tempted to not document this yet as this feels like a bug not a feature that you can connect to the queue from outside the PaaS like this... we should make a decision on what default behaviour we want before documenting it

Reply:
I'm not sure whether it's either, I recall the S3 broker had some of the plumbing restricting this (which we didn't keep). I would suggest not documenting it for now.